source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -44,7 +44,13 @@ Http::FilterHeadersStatus RoleBasedAccessControlFilter::decodeHeaders(Http::Head
   const Filters::Common::RBAC::RoleBasedAccessControlEngine& engine =
       config_->engine(callbacks_->route());
 
-  if (engine.allowed(*callbacks_->connection(), headers)) {
+  if (engine.allowed(*callbacks_->connection(), headers, true /*dark launch mode*/)) {
```

Comment:
IIRC per the C++ style guide, this should look like: `..., /*is_dark_launch=*/ true)) {`

Alternatively, use the enum you added in the protos instead so it's explicit.